### PR TITLE
linthesia 0.8.0 (new formula)

### PR DESCRIPTION
**Pull request by albfan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

relates to #93102

### CODE REVIEW

File: Formula/linthesia.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Linthesia < Formula
+  desc "Synthesia-like software for Linux"
+  homepage "http://linthesia.sourceforge.net/"
+  url "https://github.com/linthesia/linthesia/archive/refs/tags/0.8.0.tar.gz"
+  sha256 "0bbfeb30909fbd3a3b33bc45e8f7e75944ba6b2135a627d5729f4bda50540027"
+  license "GPL-2.0"
```
Comment (carlocab) [inline Formula/linthesia.rb:6]: `GPL-2.0` is deprecated. Please determine whether this is `GPL-2.0-only` or `GPL-2.0-or-later` and use the one that's appropriate.

File: Formula/linthesia.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  depends_on "gsettings-desktop-schemas" => :build
+
+  def install
+    system "meson", "--prefix=#{prefix}", "build"
+    system "ninja", "-C", "build"
+    system "ninja", "-C", "build", "install"
+  end
+
+  test do
+    system "ninja", "-C", "build", "test"
```
Comment (carlocab) [inline Formula/linthesia.rb:17]: This test will fail.

File: Formula/linthesia.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Synthesia-like software for Linux"
+  homepage "http://linthesia.sourceforge.net/"
+  url "https://github.com/linthesia/linthesia/archive/refs/tags/0.8.0.tar.gz"
+  sha256 "0bbfeb30909fbd3a3b33bc45e8f7e75944ba6b2135a627d5729f4bda50540027"
+  license "GPL-2.0"
+
+  depends_on "gsettings-desktop-schemas" => :build
+
+  def install
+    system "meson", "--prefix=#{prefix}", "build"
```
Comment (carlocab) [inline Formula/linthesia.rb:11]: Please use `std_meson_args`.

File: Formula/linthesia.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class Linthesia < Formula
+  desc "Synthesia-like software for Linux"
+  homepage "http://linthesia.sourceforge.net/"
+  url "https://github.com/linthesia/linthesia/archive/refs/tags/0.8.0.tar.gz"
+  sha256 "0bbfeb30909fbd3a3b33bc45e8f7e75944ba6b2135a627d5729f4bda50540027"
+  license "GPL-2.0"
+
+  depends_on "gsettings-desktop-schemas" => :build
```
Comment (carlocab) [inline Formula/linthesia.rb:8]: You probably need at least a `depends_on "meson" => :build` and `depends_on "ninja" => :build` here.

**Review by carlocab — CHANGES_REQUESTED** [review]
Please check out your PR branch and run
```
brew audit --new linthesia
brew test linthesia
```
as requested in the PR template. Or, at the very least, if you did not do them, please don't check the boxes saying you did.

**Comment by albfan** [comment]
I see now how test works.

```
[linuxbrew@thinkpad homebrew-core]$ ../.linuxbrew/Cellar/linthesia/0.8.0/bin/linthesia --version
../.linuxbrew/Cellar/linthesia/0.8.0/bin/linthesia: /home/linuxbrew/.linuxbrew/lib/libtiff.so.5: no version information available (required by /lib64/libSDL2_image-2.0.so.0)
ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: Permission denied
WARNING :: could not open MIDI sequencer. No MIDI available
error: XDG_RUNTIME_DIR not set in the environment.
Linthesia detected a problem and must close:

error inicializando SDL: No available video device

If you don't think this should have happened, please fill a bug report on : 
https://github.com/linthesia/linthesia

Thank you.
```

I tried adding 

    depends_on "alsa-lib"

but didn't work

**Comment by SMillerDev** [comment]
`error: XDG_RUNTIME_DIR not set in the environment.`

Have you tried setting this?

File: Formula/linthesia.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    mkdir "build" do
+      system "meson", *std_meson_args, ".."
+      system "ninja", "-v"
+      system "ninja", "install", "-v"
+    end
+  end
+
+  def post_install
+    system Formula["glib"].opt_bin/"glib-compile-schemas", HOMEBREW_PREFIX/"share/glib-2.0/schemas"
+    system Formula["gtk+3"].opt_bin/"gtk3-update-icon-cache", "-qtf", HOMEBREW_PREFIX/"share/icons/hicolor"
```
Comment (carlocab) [inline Formula/linthesia.rb:22]: This isn't going to work, since `glib` and `gtk+3` are not installed.

File: Formula/linthesia.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  test do
+    system bin/"linthesia", "--version"
```
Comment (carlocab) [inline Formula/linthesia.rb:26]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (albfan) [reply Formula/linthesia.rb:26]: This needs a midi file and will popup a GUI.

Is that ok?

Comment (carlocab) [reply Formula/linthesia.rb:26]: That's fine. Alternatively, we could try doing something with a non-existent midi file (or a non-midi file) and check whether it generates the expected error.

**Pull request closed without merging**